sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -1349,9 +1349,13 @@ private[spark] object QueryCompilationErrors {
         s"${evalTypes.mkString(",")}")
   }
 
-  def ambiguousFieldNameError(fieldName: String, names: String): Throwable = {
+  def ambiguousFieldNameError(
+      fieldName: Seq[String], nunMatches: Int, context: Origin): Throwable = {
     new AnalysisException(
-      s"Ambiguous field name: $fieldName. Found multiple columns that can match: $names")
```

Comment:
Is it
```
46121 | X | 46 | OLB-specific error | 121 | invalid column name
```
?

Reply:
I don't know if this is `OLB-specific error`. I'll pick `SQLSTATE 42000 (syntax error)` as @karenfeng suggested